### PR TITLE
BCount Better Solutions.

### PR DESCRIPTION
**Pull request by SansPapyrus683**
They all have been tested, don't worry.

### CODE REVIEW

File: solutions/USACO dec15-breed-counting.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 id: usaco-572
 source: USACO Silver 2015 December
 title: Breed Counting
-author: Óscar Garries (C++), Owen Wang (Java)
+author: Óscar Garries Owen Wang, Kevin Sheng
```
Comment (maggie-j-liu) [inline solutions/USACO dec15-breed-counting.mdx:5]:
```suggestion
author: Óscar Garries, Owen Wang, Kevin Sheng
```

File: solutions/USACO dec15-breed-counting.mdx [hunk 2]
```diff
@@ -13,37 +13,43 @@ author: Óscar Garries (C++), Owen Wang (Java)
 ## Implementation
 
 ```cpp
-#include <bits/stdc++.h>
+#include <iostream>
 
-using namespace std;
+using std::cin;
```
Comment (maggie-j-liu) [inline solutions/USACO dec15-breed-counting.mdx:18]: i don't think it's necessary to rewrite others' solutions

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/USACO dec15-breed-counting.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/USACO dec15-breed-counting.mdx:18]: that doesn't mean it's bad

Comment (andrewwangva) [inline solutions/USACO dec15-breed-counting.mdx:18]: why remove namespace std???

**Review by andrewwangva — CHANGES_REQUESTED** [review]

File: solutions/USACO dec15-breed-counting.mdx [hunk 3]
```diff
@@ -13,37 +13,43 @@ author: Óscar Garries (C++), Owen Wang (Java)
...
-		if (id == 1) h[i + 1]++;
-		else if (id == 2) g[i + 1]++;
-		else j[i + 1]++;
+	int cow_num;
+	int query_num;
+	cin >> cow_num >> query_num;
+
+	int holsteins[cow_num + 1];
+	int guernseys[cow_num + 1];
+	int jerseys[cow_num + 1];
```
Comment (thecodingwizard) [inline solutions/USACO dec15-breed-counting.mdx:31]: wait lol don't these need to be initialized to zero (at least for the zeroeth index)?

Comment (SansPapyrus683) [reply solutions/USACO dec15-breed-counting.mdx:31]: default value's 0 for java arrays

File: solutions/USACO dec15-breed-counting.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/USACO dec15-breed-counting.mdx:18]: why not lol

File: solutions/USACO dec15-breed-counting.mdx (hunk 3 above)
Comment (maggie-j-liu) [reply solutions/USACO dec15-breed-counting.mdx:31]: this isn't java...

Comment (maggie-j-liu) [reply solutions/USACO dec15-breed-counting.mdx]: also variable length arrays are nonstandard - could you change them to constant size or use vectors?

File: solutions/usaco-572.mdx [hunk 4]
```diff
@@ -52,55 +57,94 @@ int main() {
...
-		for (int i = 0; i < Q; i++) {
-			int start = sc.nextInt();
-			int end = sc.nextInt();
-
-			int Holstein = Holsteins[end] - Holsteins[start - 1];
-			int Guernsey = Guernseys[end] - Guernseys[start - 1];
-			int Jersey = Jerseys[end] - Jerseys[start - 1];
-
-			out.println(Holstein + " " + Guernsey + " " + Jersey);
+		StringBuilder ans = new StringBuilder();
```
Comment (andrewwangva) [inline solutions/usaco-572.mdx:94]: no point in using a stringbuilder - just print out the answer.

**Review by andrewwangva — CHANGES_REQUESTED** [review]

File: solutions/usaco-572.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-572.mdx:94]: if i don't use a stringbuilder it TLEs

Comment (andrewwangva) [reply solutions/usaco-572.mdx:95]: it tles when you print it out?? Dont think that's possible

Comment (SansPapyrus683) [reply solutions/usaco-572.mdx:95]: io's pretty slow in java, just trust me on this one

Comment (dongliu0426) [reply solutions/usaco-572.mdx:95]: Screen Shot 2021-05-17 at 1 07 04 PM

wdym?

Comment (SansPapyrus683) [reply solutions/usaco-572.mdx:95]: image
idk man, maybe i'm just trolling but

Comment (andrewwangva) [reply solutions/usaco-572.mdx:95]: wait im p sure printing it out is always going to be faster than stringbuilder.

Comment (andrewwangva) [reply solutions/usaco-572.mdx:95]: ^I could be wrong but nevertheless printing should always pass still

Comment (maggie-j-liu) [reply solutions/usaco-572.mdx:95]: i think it's cause @SansPapyrus683 was using printf?

Comment (SansPapyrus683) [reply solutions/usaco-572.mdx]: > wait im p sure printing it out is always going to be faster than stringbuilder.

most definitely not

**Review by andrewwangva — APPROVED** [review]

File: solutions/usaco-572.mdx [hunk 5]
```diff
@@ -52,55 +58,84 @@ int main() {
...
-				Holsteins[i] ++;
-			} else if (breed == 2) {
-				Guernseys[i] ++;
-			} else if (breed == 3) {
-				Jerseys[i] ++;
-			} else {
-				System.out.println("What the...");
-			}
+import java.io.*;
+import java.io.*;
```
Comment (maggie-j-liu) [inline solutions/usaco-572.mdx:62]:
```suggestion
```

Comment (andrewwangva) [reply solutions/usaco-572.mdx]: lol im an idiot

File: solutions/usaco-572.mdx [hunk 6]
```diff
@@ -52,55 +58,83 @@ int main() {
...
+			holsteins[i] += holsteins[i - 1];
+			guernseys[i] += guernseys[i - 1];
+			jerseys[i] += jerseys[i - 1];
+			
+			int breed = Integer.parseInt(read.readLine());
+			if(breed == 1)
+				holsteins[i]++;
+			if(breed == 2)
+				guernseys[i]++;
+			if(breed == 3)
```
Comment (maggie-j-liu) [inline solutions/usaco-572.mdx:85]:
```suggestion
			if (breed == 1)
				holsteins[i]++;
			if (breed == 2)
				guernseys[i]++;
			if (breed == 3)
```
```suggestion
			if(breed == 1)
				holsteins[i]++;
			if(breed == 2)
				guernseys[i]++;
			if(breed == 3)
```

**Review by maggie-j-liu — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

File: solutions/usaco-572.mdx [hunk 7]
```diff
@@ -52,55 +58,79 @@ int main() {
...
+			holsteins[i] += holsteins[i - 1];
+			guernseys[i] += guernseys[i - 1];
+			jerseys[i] += jerseys[i - 1];
+			
+			int breed = Integer.parseInt(read.readLine());
+			if (breed == 1)
+				holsteins[i]++;
+			if (breed == 2)
+				guernseys[i]++;
+			if (breed == 3)
```
Comment (maggie-j-liu) [inline solutions/usaco-572.mdx:85]: @andrewwangva why not else if?

Comment (andrewwangva) [reply solutions/usaco-572.mdx:85]: whoops idk i just felt it was more concise + there's no difference

Comment (maggie-j-liu) [reply solutions/usaco-572.mdx:85]: i think there is a difference?

Comment (andrewwangva) [reply solutions/usaco-572.mdx:85]: hmm how? i think its the same logically since breed can't be both 1 and 2?

Comment (maggie-j-liu) [reply solutions/usaco-572.mdx:85]: yeah, so isn't it *technically* more efficient to use else if? so breed == 2 and breed == 3 don't have to be checked if breed == 1

Comment (andrewwangva) [reply solutions/usaco-572.mdx:85]: true idk both are perfectly fine just kinda an impulsive change to make it look clean to me :)

**Pull request merged**